store/file_store.go

```diff
@@ -180,7 +180,12 @@ func (w *moveCloser) Close() error {
 	if !os.IsNotExist(err) {
 		return ErrKeyExists
 	}
-	return os.Rename(w.source, w.target)
+	err = os.Rename(w.source, w.target)
+	if err != nil {
+		return err
+	}
+	// Remove scratch directory
+	return os.RemoveAll(path.Dir(w.source))
```

Comment:
So this happens on every close. But what if another goroutine is also writing into the scratch directory?...will it be able to rename the file from the scratch directory into the proper place? It doesn't seem like that would work.

At any rate, I'm not convinced that removing the scratch directory after *every* write is worthwhile. Maybe instead add code to `butil` that specifically deletes the scratch directory (when using the FileSystem store).